apps/src/templates/lessonOverview/googleDocsUtils.js

```diff
@@ -0,0 +1,68 @@
+const GDOCS_HOST = 'docs.google.com';
+const GDOCS_URL_REGEX = /^https?:\/\/docs\.google\.com\/(document|presentation)\/d\/([\w-]+)/;
+const SLIDES_PATH_REGEX = /\/presentation\/d\/[\w-]+/;
+const DOCS_PATH_REGEX = /\/document\/d\/[\w-]+/;
+const EXPORT_PATH = '/export?format=';
+const COPY_PATH = '/copy';
+
+// Check whether the given string is a Google Docs URL in a format we
+// know how to parse. Note some valid Google Docs URLs may return `false`
+// here, if they aren't in the format we know how to interpret. The other
+// Google Docs functions below expect that the URL they're acting on has
+// already passed this function's checks.
+// Example valid URL: https://docs.google.com/document/d/SOME_LONG_ID
+export function isGDocsUrl(s) {
+  try {
+    const url = new URL(s);
+    if (url.host !== GDOCS_HOST) {
+      return false;
+    }
+
+    if (
+      SLIDES_PATH_REGEX.test(url.pathname) ||
+      DOCS_PATH_REGEX.test(url.pathname)
+    ) {
+      // The URL has the correct host, and a path we know how to parse.
+      return true;
+    }
+  } catch (err) {
+    // Expect TypeError if `s` is not a valid URL.
+    return false;
+  }
+  return false;
+}
+
+// Return a Google Docs URL in the expected format, to be appended to by
+// the more specific functions below. May raise TypeError if the URL has
+// not been checked for validity with isGDocsUrl.
```

Comment:
let me know if this seems bad. the only way anyone should be able to crash the page is if a developer passes something to one of the other functions without checking `isGDocsUrl` first. But if it seems better, I can have this return an empty string, or a `#`, or something else that doesn't cause an exception.

Reply:
i think what you have is appropriate, and if it causes issues because it's common to get malformed URLs, we can change this later. i don't think malformed URLs will be common -- our curriculum team controls those and they know that whatever URL they input will be accessed by users